core/controllers/acl_decorators.py

```diff
@@ -1822,6 +1823,81 @@ def test_can_edit_skill(self, skill_id, **kwargs):
     return test_can_edit_skill
 
 
+def can_edit_skills(handler):
+    """Decorator to check whether the user can edit multiple skills, which can
+    be independent or belong to a topic.
+
+    Args:
+        handler: function. The function to be decorated.
+
+    Returns:
+        function. The newly decorated function that now also checks if
+            the user has permission to edit skills.
+    """
+    def can_user_edit_skill(user, skill_rights):
+        """Checks whether the user can edit the given skill.
+
+        Args:
+            user: UserActionsInfo. Object having user id, role and actions for
+                given user.
+            skill_rights: SkillRights or None. Rights object for the given
+                skill.
+
+        Returns:
+            bool. Whether the given user can edit the given skill.
+        """
+        if skill_rights is None:
+            return False
+        if role_services.ACTION_EDIT_PUBLIC_SKILLS in user.actions:
+            if not skill_rights.is_private():
+                return True
+        if skill_rights.is_private() and skill_rights.is_creator(
+                user.user_id):
+            return True
+        return False
+
+    def test_can_edit_skills(self, skill_ids, **kwargs):
+        """Test to see if user can edit a given skill by checking if
+        logged in and using can_user_edit_skill.
+
+        Args:
+            skill_ids: list(str). The list of skill IDs.
+            **kwargs: *. Keyword arguments.
+
+        Returns:
+            *. The return value of the decorated function.
+
+        Raises:
+            NotLoggedInException: The user is not logged in.
+            PageNotFoundException: The given page cannot be found.
+            UnauthorizedUserException: The user does not have the
```

Comment:
There is a list of skill IDs, so I'm not sure what "the given skill" means. 

In general, please take care to ensure your docstrings are as precise as possible (I'm seeing this issue in several PRs). The current one obscures the possibility of the user having edit rights to some skills but not others.

Reply:
Done.